core/pkg/server/sender.go

```diff
@@ -1134,6 +1136,54 @@ func (s *Sender) sendSync(record *service.Record, request *service.SyncRequest)
 	s.fwdChan <- rec
 }
 
+func (s *Sender) sendStopStatus(record *service.Record, _ *service.StopStatusRequest) {
+	// TODO: ideally we could get it direclty from the settings
+	entity := s.RunRecord.GetEntity()
+	project := s.RunRecord.GetProject()
+	runId := s.RunRecord.GetRunId()
+
+	result := &service.Result{
+		ResultType: &service.Result_Response{
+			Response: &service.Response{
+				ResponseType: &service.Response_StopStatusResponse{
+					StopStatusResponse: &service.StopStatusResponse{
+						RunShouldStop: false,
+					},
+				},
+			},
+		},
+		Control: record.Control,
+		Uuid:    record.Uuid,
+	}
+
+	if entity == "" || project == "" || runId == "" {
+		s.logger.Error("sender: sendStopStatus: entity, project, runId are empty")
+		s.outChan <- result
+		return
+	}
+
+	response, err := gql.RunStoppedStatus(s.ctx,
+		s.graphqlClient,
+		&entity,
+		&project,
+		runId,
+	)
+	if err != nil {
+		err = fmt.Errorf("sender: sendStopStatus: failed to get run stopped status: %s", err)
+		s.logger.CaptureError("sender received error", err)
+		s.outChan <- result
+		return
+	}
+
+	result.ResultType.(*service.Result_Response).Response.ResponseType =
+		&service.Response_StopStatusResponse{
+			StopStatusResponse: &service.StopStatusResponse{
+				RunShouldStop: *response.GetProject().GetRun().GetStopped(),
```

Comment:
scary, this'll panic if it gets a nil. Can this be a bit more lenient and interpret `nil` as `false`?

Reply:
good point... i'll clean it up